client/signup/utils.js

```diff
@@ -216,3 +217,22 @@ export const shouldForceLogin = ( flowName, userLoggedIn ) => {
 	const flow = flows.getFlow( flowName, userLoggedIn );
 	return !! flow && flow.forceLogin;
 };
+
+// Derive if the "plans" step actually will be visible to the customer in a given flow after the domain step
+// i.e. Check "launch-site" flow while having a purchased paid plan
```

Comment:
I'd change this to JSDoc to show this description when you hover on the function in other parts of the codebase. Especially that that function's name doesn't indicate the requirement for domain step.

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/54117/commits/3c678f8876c0b46c092e0ce50a7091e7a6324437